docs/implementors-guide.md

```diff
@@ -0,0 +1,21 @@
+# Implementor's Guide
```

Comment:
feels more like "user" guide

Reply:
Yes, the `type` and `provider` details are more helpful for application authors. So, changed the title.

The implementor's guide would be targeted at the operator/reconciler implementors.